### PR TITLE
Homework3

### PR DESCRIPTION
**Pull request by kaiomv**

### CODE REVIEW

File: Src/BootCamp.Chapter/Lesson3.cs [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+            Console.WriteLine("Your BMI is " + bmiCalc(weight, height));
+
+            Console.ReadLine();
+        }
+        public static float bmiCalc(float weight, float height)
+        {
+            float bmi = (weight / (height * height));
+            return bmi;
+        }
+        public static string printAndReadString(string message)
```
Comment (Almantask) [inline Src/BootCamp.Chapter/Lesson3.cs]: Codacy Issue found: [Rename method 'printAndReadString' to match pascal case naming rules, consider using 'PrintAndReadString'.](https://app.codacy.com/manual/Almantask/CSharp-From-Zero-To-Hero/pullRequest?prid=4862789)

**Comment by Almantask** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
- Added 9
           

Complexity increasing per file
==============================
- Src/BootCamp.Chapter/Lesson3.cs  1
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/Almantask/CSharp-From-Zero-To-Hero/pullRequest?prid=4862789&bid=16129922)

File: Src/BootCamp.Chapter/Lesson3.cs [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+            int age = printAndReadInt("What is your age?");
+            float height = printAndReadFloat("What is your height in meters?");
+            float weight = printAndReadFloat("What is your weight in kg?");
+
+            Console.WriteLine(name + " " + surename + " is " + age + " years old, his weight is " + weight + " kg and his height is " + height + " m ");
+            Console.WriteLine("Your BMI is " + bmiCalc(weight, height));
+
+            Console.ReadLine();
+        }
+        public static float bmiCalc(float weight, float height)
```
Comment (Almantask) [inline Src/BootCamp.Chapter/Lesson3.cs]: Codacy Issue found: [Rename method 'bmiCalc' to match pascal case naming rules, consider using 'BmiCalc'.](https://app.codacy.com/manual/Almantask/CSharp-From-Zero-To-Hero/pullRequest?prid=4862789)

File: Src/BootCamp.Chapter/Checks.cs [hunk 3]
```diff
@@ -14,28 +16,32 @@ namespace BootCamp.Chapter
     /// </summary>
     public static class Checks
     {
+        
         public static int PromptInt(string message)
         {
+            
             // To do: call your implementation. 
-            return 0;
+            return Convert.ToInt32(PromptString(message)); ;
```
Comment (Almantask) [inline Src/BootCamp.Chapter/Checks.cs]: Codacy Issue found: [Remove this empty statement.](https://app.codacy.com/manual/Almantask/CSharp-From-Zero-To-Hero/pullRequest?prid=4862789)

File: Src/BootCamp.Chapter/Lesson3.cs [hunk 4]
```diff
@@ -0,0 +1,42 @@
...
+        {
+            float bmi = (weight / (height * height));
+            return bmi;
+        }
+        public static string printAndReadString(string message)
+        {
+            Console.WriteLine(message);
+            return Console.ReadLine();
+        }
+        public static int printAndReadInt(string message)
```
Comment (Almantask) [inline Src/BootCamp.Chapter/Lesson3.cs]: Codacy Issue found: [Rename method 'printAndReadInt' to match pascal case naming rules, consider using 'PrintAndReadInt'.](https://app.codacy.com/manual/Almantask/CSharp-From-Zero-To-Hero/pullRequest?prid=4862789)

File: Src/BootCamp.Chapter/Lesson3.cs [hunk 5]
```diff
@@ -0,0 +1,42 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Text;
```
Comment (Almantask) [inline Src/BootCamp.Chapter/Lesson3.cs]: Codacy Issue found: [Remove this unnecessary 'using'.](https://app.codacy.com/manual/Almantask/CSharp-From-Zero-To-Hero/pullRequest?prid=4862789)

File: Src/BootCamp.Chapter/Lesson3.cs [hunk 6]
```diff
@@ -0,0 +1,42 @@
...
+        public static string printAndReadString(string message)
+        {
+            Console.WriteLine(message);
+            return Console.ReadLine();
+        }
+        public static int printAndReadInt(string message)
+        {
+            return Convert.ToInt32(printAndReadString(message));
+        }
+        public static float printAndReadFloat(string message)
```
Comment (Almantask) [inline Src/BootCamp.Chapter/Lesson3.cs]: Codacy Issue found: [Add a way to break out of this method's recursion.](https://app.codacy.com/manual/Almantask/CSharp-From-Zero-To-Hero/pullRequest?prid=4862789)

Comment (Almantask) [inline Src/BootCamp.Chapter/Lesson3.cs]: Codacy Issue found: [Rename method 'printAndReadFloat' to match pascal case naming rules, consider using 'PrintAndReadFloat'.](https://app.codacy.com/manual/Almantask/CSharp-From-Zero-To-Hero/pullRequest?prid=4862789)

File: Src/BootCamp.Chapter/Lesson3.cs [hunk 7]
```diff
@@ -0,0 +1,42 @@
+﻿using System;
+using System.Collections.Generic;
```
Comment (Almantask) [inline Src/BootCamp.Chapter/Lesson3.cs]: Codacy Issue found: [Remove this unnecessary 'using'.](https://app.codacy.com/manual/Almantask/CSharp-From-Zero-To-Hero/pullRequest?prid=4862789)

File: Src/BootCamp.Chapter/Program.cs [hunk 8]
```diff
@@ -1,6 +1,7 @@
 ﻿using System;
 using System.Collections.Generic;
 using System.IO;
+using System.Globalization;
```
Comment (Almantask) [inline Src/BootCamp.Chapter/Program.cs]: Codacy Issue found: [Remove this unnecessary 'using'.](https://app.codacy.com/manual/Almantask/CSharp-From-Zero-To-Hero/pullRequest?prid=4862789)

File: Src/BootCamp.Chapter/Program.cs [hunk 9]
```diff
@@ -9,6 +10,14 @@ class Program
...
         }
+
+
+
+        
     }
 }
+    
+ 
+
```
Comment (ellmaster) [inline Src/BootCamp.Chapter/Program.cs]: Please remove those empty lines. :)

File: Src/BootCamp.Chapter/Lesson3.cs [hunk 10]
```diff
@@ -0,0 +1,42 @@
...
+{
+    class Lesson3
+    {
+        static void Main(string[] args)
+        {
+            string name = printAndReadString("What is your name?");
+            string surename = printAndReadString("What is your surename?");
+            int age = printAndReadInt("What is your age?");
+            float height = printAndReadFloat("What is your height in meters?");
+            float weight = printAndReadFloat("What is your weight in kg?");
```
Comment (ellmaster) [inline Src/BootCamp.Chapter/Lesson3.cs]: Convention for function names is PascalCase. Please adjust that.

File: Src/BootCamp.Chapter/Checks.cs [hunk 11]
```diff
@@ -14,28 +16,32 @@ namespace BootCamp.Chapter
...
-            return 0;
+            return Convert.ToSingle(PromptString(message), CultureInfo.InvariantCulture);
         }
 
         public static float CalculateBmi(float weight, float height)
         {
             // To do: call your implementation. 
-            return 0;
+            float bmi = (weight / (height * height));
+            return bmi;
```
Comment (ellmaster) [inline Src/BootCamp.Chapter/Checks.cs]: You shouldn't write duplicated code. Use your own functions - as the TODO comments suggest.
After using your own functions please delete those comments. :)

File: Src/BootCamp.Chapter/Program.cs [hunk 12]
```diff
@@ -9,6 +10,14 @@ class Program
     {
         static void Main(string[] args)
         {
+            Lesson3.Demo();
```
Comment (ellmaster) [inline Src/BootCamp.Chapter/Program.cs]: This function doesn't exist. You have to create it first!

File: Src/BootCamp.Chapter/Lesson3.cs [hunk 13]
```diff
@@ -0,0 +1,42 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Text;
+using System.Globalization;
+
+namespace BootCamp.Chapter
+{
+    class Lesson3
+    {
+        static void Main(string[] args)
```
Comment (ellmaster) [inline Src/BootCamp.Chapter/Lesson3.cs]: You can only have one Main function. Look in the [bc-homework](https://discordapp.com/channels/475671343463923714/663436225255964697/668202071547183104) channel which file is supposed to have that Main function.

**Comment by Almantask** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
- Added 1
           

Complexity increasing per file
==============================
- Src/BootCamp.Chapter/Lesson3.cs  1
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/Almantask/CSharp-From-Zero-To-Hero/pullRequest?prid=4862789&bid=16129922)

File: Src/BootCamp.Chapter/Checks.cs [hunk 14]
```diff
@@ -14,28 +16,26 @@ namespace BootCamp.Chapter
     /// </summary>
     public static class Checks
     {
+        
         public static int PromptInt(string message)
         {
-            // To do: call your implementation. 
-            return 0;
+            return Convert.ToInt32(PromptString(message)); ;
```
Comment (Almantask) [inline Src/BootCamp.Chapter/Checks.cs]: Codacy Issue found: [Remove this empty statement.](https://app.codacy.com/manual/Almantask/CSharp-From-Zero-To-Hero/pullRequest?prid=4862789)

**Comment by Almantask** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
- Added 2
           

Complexity increasing per file
==============================
- Src/BootCamp.Chapter/Lesson3.cs  1
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/Almantask/CSharp-From-Zero-To-Hero/pullRequest?prid=4862789&bid=16129922)

File: Src/BootCamp.Chapter/Checks.cs (hunk 14 above)
Comment (Almantask) [inline Src/BootCamp.Chapter/Checks.cs]: Codacy Issue found: [Remove this empty statement.](https://app.codacy.com/manual/Almantask/CSharp-From-Zero-To-Hero/pullRequest?prid=4862789)

File: Src/BootCamp.Chapter/Checks.cs [hunk 15]
```diff
@@ -1,9 +1,11 @@
 ﻿using System;
 using System.Collections.Generic;
 using System.Text;
+using System.Globalization;
```
Comment (Almantask) [inline Src/BootCamp.Chapter/Checks.cs]: Codacy Issue found: [Remove this unnecessary 'using'.](https://app.codacy.com/manual/Almantask/CSharp-From-Zero-To-Hero/pullRequest?prid=4862789)

**Comment by Almantask** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Complexity increasing per file
==============================
- Src/BootCamp.Chapter/Lesson3.cs  1
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/Almantask/CSharp-From-Zero-To-Hero/pullRequest?prid=4862789&bid=16129922)

File: Src/BootCamp.Chapter/Checks.cs [hunk 16]
```diff
@@ -19,29 +18,23 @@ public static class Checks
         
         public static int PromptInt(string message)
         {
-            
-            // To do: call your implementation. 
-            return Convert.ToInt32(PromptString(message)); ;
+            return Convert.ToInt32(PromptString(message));
```
Comment (ellmaster) [inline Src/BootCamp.Chapter/Checks.cs]: Those 4 functions should return something that looks like `Lesson3.Function(message)` :)

File: Src/BootCamp.Chapter/Lesson3.cs [hunk 17]
```diff
@@ -1,42 +1,38 @@
...
+        public static string PrintAndReadString(string message)
         {
             Console.WriteLine(message);
             return Console.ReadLine();
         }
-        public static int printAndReadInt(string message)
+        public static int PrintAndReadInt(string message)
         {
-            return Convert.ToInt32(printAndReadString(message));
+            return Convert.ToInt32(PrintAndReadString(message));
```
Comment (ellmaster) [inline Src/BootCamp.Chapter/Lesson3.cs]: I think this is a beautiful idea! I like it very much.

**Comment by Almantask** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Complexity increasing per file
==============================
- Src/BootCamp.Chapter/Lesson3.cs  1
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/Almantask/CSharp-From-Zero-To-Hero/pullRequest?prid=4862789&bid=16129922)

**Comment by Almantask** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Complexity increasing per file
==============================
- Src/BootCamp.Chapter/Lesson3.cs  1
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/Almantask/CSharp-From-Zero-To-Hero/pullRequest?prid=4862789&bid=16129922)

File: Src/BootCamp.Chapter/Lesson3.cs [hunk 18]
```diff
@@ -0,0 +1,47 @@
+﻿using System;
+
+namespace BootCamp.Chapter
+{
+    class Lesson3
+    {
+        public static void Demo()
+        {
+            string name = PrintAndReadString("Hello, what's the name of the person that you want to calculate the BMI?");
```
Comment (Almantask) [inline Src/BootCamp.Chapter/Lesson3.cs]: Use a function to remove code duplications for the same person info reads.

**Comment by Almantask** [comment]
Expired

**Pull request closed without merging**